.github/workflows/pull_request.workflow.yml

```diff
@@ -0,0 +1,156 @@
+name: Pull request checks
+
+on: [pull_request]
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+      - name: Cache node modules
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-node-modules
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+      - uses: actions/setup-node@v1
+        with:
+          node-version: "12"
+      - uses: ./.github/actions/es-lint
+
+  unit-tests:
+    name: Unit Tests - Node.js v${{ matrix.node-version }}
+    needs: ['lint']
+    strategy:
+      matrix:
+        node-version: ['12.16', '12.18']
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+      - name: Cache node modules
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-node-modules
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+      - uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - uses: ./.github/actions/unit-tests
+
+  functional-tests-legacy:
+    name: Functional Tests Legacy - ${{ matrix.test_set }} - Node.js v${{ matrix.node-version }}
+    needs: ['unit-tests']
+    strategy:
+      matrix:
+        test_set: ['http', 'websocket', 'mqtt']
+        node-version: ['12.16', '12.18']
```

Comment:
Functional tests does not use the host Node version, the `NODE_VERSION` variable must be set to `12.16.3` for 12.16 tests. By default it's set on 12.18 but this behavior should be avoided since it's not explicit